entity_detection/nn/retrieval.py

```diff
@@ -0,0 +1,144 @@
+import torch
```

Comment:
can we rename this file to 'top_retrieval.py' to be consistent with the relation prediction directory